client/client-multi/client-cpp/impl/channel/impl/DefaultOperationTcpChannel.cpp

```diff
@@ -751,39 +793,24 @@ void DefaultOperationTcpChannel::shutdown()
 
 void DefaultOperationTcpChannel::pause()
 {
-    KAA_MUTEX_LOCKING("channelGuard_");
-    KAA_MUTEX_UNIQUE_DECLARE(lock, channelGuard_);
-    KAA_MUTEX_LOCKED("channelGuard_");
-
+    std::lock_guard<std::recursive_mutex> lock(channelGuard_);
     if (isShutdown_) {
         KAA_LOG_WARN(boost::format("Channel [%1%] can't pause: channel is shut down") % getId());
         return;
     }
 
-    if (!isPaused_) {
-        isPaused_ = true;
-        KAA_MUTEX_UNLOCKING("channelGuard_");
-        KAA_UNLOCK(lock);
-        KAA_MUTEX_UNLOCKED("channelGuard_");
-        closeConnection();
-    }
+    io_.post(std::bind(&DefaultOperationTcpChannel::closeConnection, this));
```

Comment:
Not sure why that should be called on `io_`. Can't `ChannelConnection` be safely destroyed from any thread?